pkg/daemon/daemon.go

```diff
@@ -214,22 +215,34 @@ func (dn *Daemon) addProfileConfig(socketPath string, nodeProfile *ptpv1.PtpProf
 		nodeProfile.Ptp4lOpts = &opts
 	}
 
+	output := &ptp4lConf{}
 	if nodeProfile.Interface == nil || *nodeProfile.Interface == "" {
-		// We need to copy this to another variable because is a pointer
 		iface := string("")
 		nodeProfile.Interface = &iface
+	} else {
+		output.ifaceString = fmt.Sprintf("[%s]", *nodeProfile.Interface)
 	}
 
-	config := fmt.Sprintf("%s\nuds_address %s\nmessage_tag [%s]",
-		*nodeProfile.Ptp4lConf,
-		socketPath,
-		*nodeProfile.Interface)
-	nodeProfile.Ptp4lConf = &config
+	output.populatePtp4lConf(nodeProfile.Ptp4lConf)
+
+	if nodeProfile.Interface != nil && *nodeProfile.Interface != "" {
+		ifaceSection := fmt.Sprintf("[%s]\n", *nodeProfile.Interface)
+		output.sections[ifaceSection] = *new(ptp4lConfSection)
+	}
+
+	section := output.sections["[global]"]
+	section.options["message_tag"] = fmt.Sprintf("[%s]", configFile)
+	section.options["uds_address"] = socketPath
+	output.sections["[global]"] = section
+
+	output.renderPtp4lConf()
```

Comment:
lets make this function to return a string no need to save it in the conf variable

Reply:
I need to update to do mapping when rendering in order to pick up ptpconfig interface in interface list, so need to use conf variable anyways.